File: back/domain/inputs/instalacion.go
package inputs

type InstalacionInput struct {
	Tipo             string   `json:"tipo"`
	Nombre           string   `json:"nombre"`
	Estado           string   `json:"estado"`
	Lat              float64  `json:"lat"`
	Lng              float64  `json:"lng"`
	Direccion        string   `json:"direccion"`
	Zona             string   `json:"zona"`
	FechaInstalacion string   `json:"fecha_instalacion"`
	Serial           string   `json:"serial"`
	Descripcion      *string  `json:"descripcion,omitempty"`
	Fabricante       *string  `json:"fabricante,omitempty"`
	Modelo           *string  `json:"modelo,omitempty"`
	Instalador       *string  `json:"instalador,omitempty"`
	Proveedor        *string  `json:"proveedor,omitempty"`
	UltimaConexion   *string  `json:"ultima_conexion,omitempty"`
	Firmware         *string  `json:"firmware,omitempty"`
	IP               *string  `json:"ip,omitempty"`
	MAC              *string  `json:"mac,omitempty"`
	Resolucion       *string  `json:"resolucion,omitempty"`
	Contenido        *string  `json:"contenido,omitempty"`
	Senal            *float64 `json:"senal,omitempty"`
	Bateria          *float64 `json:"bateria,omitempty"`
	Potencia         *float64 `json:"potencia,omitempty"`
	Contacto         *string  `json:"contacto,omitempty"`
}

type InstalacionUpdateInput struct {
	Tipo             *string   `json:"tipo,omitempty"`
	Nombre           *string   `json:"nombre,omitempty"`
	Estado           *string   `json:"estado,omitempty"`
	Lat              *float64  `json:"lat,omitempty"`
	Lng              *float64  `json:"lng,omitempty"`
	Direccion        *string   `json:"direccion,omitempty"`
	Zona             *string   `json:"zona,omitempty"`
	FechaInstalacion *string   `json:"fecha_instalacion,omitempty"`
	Serial           *string   `json:"serial,omitempty"`
	Descripcion      *string   `json:"descripcion,omitempty"`
	Fabricante       *string   `json:"fabricante,omitempty"`
	Modelo           *string   `json:"modelo,omitempty"`
	Instalador       *string   `json:"instalador,omitempty"`
	Proveedor        *string   `json:"proveedor,omitempty"`
	UltimaConexion   **string  `json:"ultima_conexion,omitempty"`
	Firmware         *string   `json:"firmware,omitempty"`
	IP               *string   `json:"ip,omitempty"`
	MAC              *string   `json:"mac,omitempty"`
	Resolucion       *string   `json:"resolucion,omitempty"`
	Contenido        *string   `json:"contenido,omitempty"`
	Senal            **float64 `json:"senal,omitempty"`
	Bateria          **float64 `json:"bateria,omitempty"`
	Potencia         **float64 `json:"potencia,omitempty"`
	Contacto         *string   `json:"contacto,omitempty"`
}

File: front/src/app/models/instalacion.ts
export const INSTALACION_TIPOS = ['CAMARA', 'PANEL', 'BANNER', 'SENSOR', 'ROUTER'] as const;
export type InstalacionTipo = (typeof INSTALACION_TIPOS)[number];

export const INSTALACION_ESTADOS = [
  'ACTIVO',
  'INACTIVO',
  'MANTENIMIENTO',
  'MALOGRADO',
  'PENDIENTE_INSTALACION',
  'PENDIENTE_REPARACION',
] as const;
export type InstalacionEstado = (typeof INSTALACION_ESTADOS)[number];

export const INSTALACION_ZONAS = ['CENTRO', 'NORTE', 'SUR', 'ESTE', 'OESTE'] as const;
export type InstalacionZona = (typeof INSTALACION_ZONAS)[number];

export interface Instalacion {
  id: number;
  tipo: InstalacionTipo;
  nombre: string;
  estado: InstalacionEstado;
  lat: number;
  lng: number;
  direccion: string;
  zona: InstalacionZona;
  fecha_instalacion: string;
  serial: string;

  descripcion?: string;
  fabricante?: string;
  modelo?: string;
  instalador?: string;
  proveedor?: string;
  ultima_conexion?: string | null;
  firmware?: string;
  ip?: string;
  mac?: string;
  resolucion?: string;
  contenido?: string;
  senal?: number;
  bateria?: number;
  potencia?: number;
  contacto?: string;
}

export const ESTADO_CONFIG: Record<InstalacionEstado, { label: string; color: string; background: string }> = {
  ACTIVO: { label: 'Activo', color: '#15803d', background: '#dcfce7' },
  INACTIVO: { label: 'Inactivo', color: '#475569', background: '#e2e8f0' },
  MANTENIMIENTO: { label: 'Mantenimiento', color: '#b45309', background: '#fef3c7' },
  MALOGRADO: { label: 'Malogrado', color: '#b91c1c', background: '#fee2e2' },
  PENDIENTE_INSTALACION: { label: 'Pend. instalación', color: '#1d4ed8', background: '#dbeafe' },
  PENDIENTE_REPARACION: { label: 'Pend. reparación', color: '#c2410c', background: '#ffedd5' },
};